### PR TITLE
am: add section about SAML 2.0 protocol

### PR DESCRIPTION
**Pull request by tcompiegne**
closes gravitee-io/issues#7011

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://graviteedocs.blob.core.windows.net/issues-7011-saml2-idp/index.html)

### CODE REVIEW

File: pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+- Web Browser SSO Profile
+- Single Logout Profile
+- SP-Initiated flow
+- Support for signed SAML assertions (SAML Request and SAML Response)
+
+NOTE: Support for encrypted SAML assertions will be provided in a next version of the SAML 2.0 IdP protocol plugin.
+
+== Activate SAML 2.0 IdP
+
+NOTE: Be sure to have your SAML 2.0 IdP protocol plugin and your license key installed in your environment before configuring the connection.
```
Comment (leleueri) [inline pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc:32]: Do we have to specify on which service ?
Currently only the GW need such installation.

File: pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+- *SingleSignOnService*, the SAML IdP Sign In URL : https://AM_GATEWAY/{domain}/saml2/idp/SSO
+- *SingleLogoutService*, the SAML IdP Sign Out URL : https://AM_GATEWAY/{domain}/saml2/idp/SSO
+- *Signing certificate*, the public signing certificate (encoded in PEM)
+
+NOTE: SAML IdP metadata information can be found here : https://AM_GATEWAY/{domain}/saml2/idp/metadata
+
+You can test your SAML 2.0 connection using a web application created in AM.
+
+. In AM Console, create a new web application.
```
Comment (leleueri) [inline pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc:55]: Show we describe how to configure the Application Part into the `Test Connection` section :thinking: ? 
Maybe we should have a section dedicated to the Application definition (on the SAML IDP server side) and into the Test connection section only explain that we can create another OAuth App (into the same domain or in another one) that use the IdentityProvider SAML plugin ?

**Review by leleueri — CHANGES_REQUESTED** [review]

File: pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc (hunk 1 above)
Comment (tcompiegne) [reply pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc:32]: I don't know to be honest

File: pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+. Click *Save*.
+
+NOTE: If you choose to not use certificate, the SAML Response assertion will not be signed.
+
+== Test the connection
+
+To connect your applications to the AM SAML 2.0 IdP, you need at least the following information:
+
+- *SingleSignOnService*, the SAML IdP Sign In URL : https://AM_GATEWAY/{domain}/saml2/idp/SSO
+- *SingleLogoutService*, the SAML IdP Sign Out URL : https://AM_GATEWAY/{domain}/saml2/idp/SSO
```
Comment (tcompiegne) [inline pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc:48]: logout

Comment (tcompiegne) [reply pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc]: FP

File: pages/am/3.x/dev-guide/protocols/protocols-overview.adoc [hunk 4]
```diff
@@ -21,4 +21,6 @@ Its intent is to reduce the cost and complexity of user management operations by
 
 - link:{{ '/am/current/am_devguide_protocols_fapi_overview.html' | relative_url }}[FAPI 1.0]: The Financial-grade API (FAPI) is a highly secured OAuth profile that aims to provide specific implementation guidelines for security and interoperability.
 
-- link:{{ '/am/current/am_devguide_protocols_ciba_overview.html' | relative_url }}[CIBA 1.0]: The Client-Initiated Backchannel Authentication Flow - Core 1.0 (CIBA) is an authentication flow where the Relying Party communicates with OpenID Provider without redirects through the user's browser.
\ No newline at end of file
+- link:{{ '/am/current/am_devguide_protocols_ciba_overview.html' | relative_url }}[CIBA 1.0]: The Client-Initiated Backchannel Authentication Flow - Core 1.0 (CIBA) is an authentication flow where the Relying Party communicates with OpenID Provider without redirects through the user's browser.
```
Comment (farmborough) [inline pages/am/3.x/dev-guide/protocols/protocols-overview.adoc:24]: "communicates with **an** OpenID Provider"

File: pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+NOTE: Be sure to have your SAML 2.0 IdP protocol plugin and your license key installed in your environment before configuring the connection.
+
+. Log in to AM Console.
+. Click *Settings > SAML 2.0*.
+. Enable *SAML 2.0 IdP support*.
+. Enter your IdP Entity ID.
+. Select your certificate to sign the SAML Response assertion.
+. Click *Save*.
+
+NOTE: If you choose to not use certificate, the SAML Response assertion will not be signed.
```
Comment (farmborough) [inline pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc:41]: "**a** certificate"

File: pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+You can test your SAML 2.0 connection using a web application created in AM.
+
+. In AM Console, create a new web application.
+. Click *Settings > SAML 2.0*.
+. Verify / update the SAML 2.0 application settings.
+. Select an identity provider to connect your users.
+. Call the Login page (the /saml/idp/SSO?SAMLRequest=... endpoint).
+. Enter username/password and click *Sign in*.
+. If everything OK, your user will be redirected to the application *attribute consume service URL* with the SAML Response assertion as a parameter.
```
Comment (farmborough) [inline pages/am/3.x/dev-guide/protocols/saml2/protocols_saml2_configuration.adoc:61]: "If everything **is** OK"

**Pull request merged**